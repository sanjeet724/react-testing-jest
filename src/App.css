.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 100px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.task-container-todo {
  display: inline-block;
  margin: 20px;
}

.task-container-done {
  display: inline-block;
  margin: 20px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
