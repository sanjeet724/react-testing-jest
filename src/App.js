import React, { Component } from 'react';
import './App.css';
import TaskItems from './components/taskItems'
import DoneItems from './components/doneItems'
import './components/listStyles.css'
const State = require('./State');

class App extends Component {
  state = State.getInitialState();

  addTask = () => {
    if (this.state.toAdd.text) {
      // add to the list
      const tasks = [...this.state.tasks];
      const newTask = {
        id: tasks.length + 1,
        text: this.state.toAdd.text,
        status: "NOK"
      }
      tasks.push(newTask);
      this.setState({ tasks });

      // clear the input
      const toAdd = { ...this.state.toAdd };
      toAdd.text = "";
      this.setState({ toAdd });
    }
  };

  handleChange = (e) => {
    if (e.target.value) {
      const toAdd = { ...this.state.toAdd };
      toAdd.text = e.target.value;
      this.setState({ toAdd });
    }
  };

  handleKeyPress = (e) => {
    if (e.key === "Enter") {
      const toAdd = { ...this.state.toAdd };
      toAdd.text = e.target.value;
      this.setState({ toAdd });
      this.addTask();
    }
  };

  handleTaskStatusChange = (task) => {
    const tasks = [...this.state.tasks];
    const index = tasks.indexOf(task);
    tasks[index].status = "OK";

    this.setState({ tasks });
  };

  render() {
    return (
      <div className="App container">
        <h1 className="App-header">My To-Do App</h1>
        <div className="input-group">
          <div className="input-group-prepend">
            <button onClick={this.addTask} className="btn btn-primary m-2">Add Something</button>
          </div>
          <input type="text" className="form-control m-2" maxLength="30"
            placeholder={this.state.toAdd.placeholder}
            value={this.state.toAdd.text}
            onChange={(event) => this.handleChange(event)}
            onKeyPress={(event) => this.handleKeyPress(event)} />
        </div>
        <div className="task-container-todo">
          <TaskItems
            tasks={this.state.tasks.filter(task => task.status === "NOK")}
            onTaskCompleted={this.handleTaskStatusChange} />
        </div>
        <div className="task-container-done">
          <DoneItems tasks={this.state.tasks.filter(task => task.status === "OK")} />
        </div>
      </div>
    );
  }
}

export default App;
