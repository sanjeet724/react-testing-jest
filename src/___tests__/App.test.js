import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';
import * as TestUtils from 'react-dom/test-utils';


// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("The App", () => {
  xit("should initialize the app correctly", () => {
    const myApp = new App();
    expect(myApp).toBeDefined();
    expect(myApp.state).toBeDefined();
    expect(myApp.state.tasks.length).toEqual(3);
  });

  it("can add a new task", () => {
    // render the app
    const myApp = TestUtils.renderIntoDocument(<App />);
    // create a new task
    let toAdd = { ...myApp.state };
    toAdd.text = "adding a new task - test";
    myApp.setState({ toAdd });
    // add the task
    myApp.addTask();
    expect(myApp.state.tasks.length).toEqual(4);
  });
});

// snapshot the app
describe("App Component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});



