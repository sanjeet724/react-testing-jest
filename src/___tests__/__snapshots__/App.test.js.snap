// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<div
  className="App container"
>
  <h1
    className="App-header"
  >
    My To-Do App
  </h1>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <button
        className="btn btn-primary m-2"
        onClick={[Function]}
      >
        Add Something
      </button>
    </div>
    <input
      className="form-control m-2"
      maxLength="30"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="What do you want to do ?"
      type="text"
      value=""
    />
  </div>
  <div
    className="task-container-todo"
  >
    <div
      className="list-group"
    >
      <h5
        className="undone-list-header"
      >
        Up Coming
      </h5>
      <div>
        <li
          className="list-group-item undone-list-item list-group-item-primary"
        >
          Start investing
          <input
            className="m-2 undone-list-checkbox"
            onClick={[Function]}
            type="checkbox"
          />
        </li>
      </div>
      <div>
        <li
          className="list-group-item undone-list-item list-group-item-primary"
        >
          Buy Apple Watch 4
          <input
            className="m-2 undone-list-checkbox"
            onClick={[Function]}
            type="checkbox"
          />
        </li>
      </div>
      <div>
        <li
          className="list-group-item undone-list-item list-group-item-primary"
        >
          Dev talk on JEST
          <input
            className="m-2 undone-list-checkbox"
            onClick={[Function]}
            type="checkbox"
          />
        </li>
      </div>
    </div>
  </div>
  <div
    className="task-container-done"
  >
    <div
      className="list-group"
    >
      <h5
        className="done-list-header"
      >
        Done
      </h5>
      <div>
        <li
          className="list-group-item done-list-item list-group-item-success"
        >
          Go to the gym
        </li>
      </div>
      <div>
        <li
          className="list-group-item done-list-item list-group-item-success"
        >
          Learn JEST
        </li>
      </div>
    </div>
  </div>
</div>
`;
